arangod/RestServer/QueryRegistryFeature.cpp

```diff
@@ -159,6 +159,16 @@ void QueryRegistryFeature::validateOptions(std::shared_ptr<ProgramOptions> optio
 
   // cap the value somehow. creating this many plans really does not make sense
   _maxQueryPlans = std::min(_maxQueryPlans, decltype(_maxQueryPlans)(1024));
+  
+  if (_queryRegistryTTL <= 0) {
+    // set to default value based on instance type
```

Comment:
Is the server role already known in `validateOptions()`? 
I think so, because the server role is set in `ClusterFeature::validateOptions()`, and the QueryRegistryFeature comes after the ClusterFeature in the defined ordering.

Reply:
I added an assertion